metadata-ingestion/scripts/avro_codegen.py

```diff
@@ -333,6 +344,321 @@ class AspectBag(TypedDict, total=False):
+    "corpGroup": [_create_from_id.format(class_name="CorpGroupUrn")],
+    "corpuser": [_create_from_id.format(class_name="CorpUserUrn")],
+    "dataFlow": [
+        """
+@classmethod
+def create_from_ids(
+    cls,
+    orchestrator: str,
+    flow_id: str,
+    env: str,
+    platform_instance: Optional[str] = None,
+) -> "DataFlowUrn":
+    return cls(
+        orchestrator=orchestrator,
+        flow_id=f"{platform_instance}.{flow_id}" if platform_instance else flow_id,
+        cluster=env,
+    )
+
+@deprecated(reason="Use .orchestrator instead")
+def get_orchestrator_name(self) -> str:
+    return self.orchestrator
+
+@deprecated(reason="Use .flow_id instead")
+def get_flow_id(self) -> str:
+    return self.flow_id
+
+@deprecated(reason="Use .cluster instead")
+def get_env(self) -> str:
+    return self.cluster
+""",
+    ],
+    "dataJob": [
+        """
+@classmethod
+def create_from_ids(cls, data_flow_urn: str, job_id: str) -> "DataJobUrn":
+    return cls(data_flow_urn, job_id)
+
+def get_data_flow_urn(self) -> "DataFlowUrn":
+    return DataFlowUrn.from_string(self.flow)
+
+@deprecated(reason="Use .job_id instead")
+def get_job_id(self) -> str:
+    return self.job_id
+"""
+    ],
+    "dataPlatform": [_create_from_id.format(class_name="DataPlatformUrn")],
+    "dataProcessInstance": [
+        _create_from_id.format(class_name="DataProcessInstanceUrn"),
+        """
+@deprecated(reason="Use .id instead")
+def get_dataprocessinstance_id(self) -> str:
+    return self.id
+""",
+    ],
+    "dataset": [
+        """
+@classmethod
+def create_from_ids(
+    cls,
+    platform_id: str,
+    table_name: str,
+    env: str,
+    platform_instance: Optional[str] = None,
+) -> "DatasetUrn":
+    return DatasetUrn(
+        platform=platform_id,
+        name=f"{platform_instance}.{table_name}" if platform_instance else table_name,
+        env=env,
+    )
+
+from datahub.utilities.urns.field_paths import get_simple_field_path_from_v2_field_path as _get_simple_field_path_from_v2_field_path
+
+get_simple_field_path_from_v2_field_path = staticmethod(deprecated(reason='Use the function from the field_paths module instead')(_get_simple_field_path_from_v2_field_path))
+
+def get_data_platform_urn(self) -> "DataPlatformUrn":
+    return DataPlatformUrn.from_string(self.platform)
+
+@deprecated(reason="Use .name instead")
+def get_dataset_name(self) -> str:
+    return self.name
+
+@deprecated(reason="Use .env instead")
+def get_env(self) -> str:
+    return self.env
+"""
+    ],
+    "domain": [_create_from_id.format(class_name="DomainUrn")],
+    "notebook": [
+        """
+@deprecated(reason="Use .notebook_tool instead")
+def get_platform_id(self) -> str:
+    return self.notebook_tool
+
+@deprecated(reason="Use .notebook_id instead")
+def get_notebook_id(self) -> str:
+    return self.notebook_id
+"""
+    ],
+    "tag": [_create_from_id.format(class_name="TagUrn")],
+}
+
+
+def generate_urn_class(entity_type: str, key_aspect: dict) -> str:
```

Comment:
This method got too hard for me to follow lol, without seeing the end product

Reply:
added a comment